roles/openshift_facts/library/openshift_facts.py

```diff
@@ -534,6 +534,27 @@ def set_aggregate_facts(facts):
 
     return facts
 
+def set_openstack_facts(facts):
+    """ Set openstack facts
+
+        Args:
+            facts (dict): existing facts
+        Returns:
+            dict: the facts dict updated with openstack facts
+    """
+
+    def _add_openstack_config(conf):
+        conf.setdefault('cloud-provider', []).append('openstack')
+        conf.setdefault('cloud-config', []).append('/etc/cloud.conf')
+
+    if 'master' in facts:
+        _add_openstack_config(facts['master'].setdefault('api_server_args', {}))
+        _add_openstack_config(facts['master'].setdefault('controller_args', {}))
+
+    if 'node' in facts:
+        _add_openstack_config(facts['node'].setdefault('kubelet_args', {}))
+
+    return facts
```

Comment:
Could we not leverage the detection we are using in guess_host_provider to do this?  We could store the default configs above in the facts['provider'] key and then all that should be needed is to merge the *_args from the provider info with the *_args in facts after the user defined changes are applied.

That should make it easier to leverage the config for the other cloud providers as well.